pkg/process/net/resolver/resolver.go

```diff
@@ -12,20 +12,71 @@ import (
 	"sync"
 	"time"
 
+	"github.com/benbjohnson/clock"
+
 	model "github.com/DataDog/agent-payload/v5/process"
 
 	procutil "github.com/DataDog/datadog-agent/pkg/process/util"
+	proccontainers "github.com/DataDog/datadog-agent/pkg/process/util/containers"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 const defaultTTL = 10 * time.Second
+const cacheValidityNoRT = 2 * time.Second
 
 // LocalResolver is responsible resolving the raddr of connections when they are local containers
 type LocalResolver struct {
-	mux         sync.RWMutex
-	addrToCtrID map[model.ContainerAddr]string
-	ctrForPid   map[int]string
-	updated     time.Time
+	mux                sync.RWMutex
+	addrToCtrID        map[model.ContainerAddr]string
+	ctrForPid          map[int]string
+	updated            time.Time
+	lastContainerRates map[string]*proccontainers.ContainerRateMetrics
+	Clock              clock.Clock
+	ContainerProvider  proccontainers.ContainerProvider
+	done               chan bool
+}
+
+func NewLocalResolver(containerProvider proccontainers.ContainerProvider, clock clock.Clock) *LocalResolver {
+	return &LocalResolver{
+		ContainerProvider: containerProvider,
+		Clock:             clock,
+		done:              make(chan bool),
+	}
+}
+
+func (l *LocalResolver) Run() {
+	pullContainerFrequency := 10 * time.Second
```

Comment:
Any reason not to add this as a const along with the `defaultTTL` and `cacheValidityNoRT`?